pkg/odp/segment/segment_option.go

```diff
@@ -17,12 +17,36 @@
 // Package segment //
 package segment
 
+import "errors"
+
 // OptimizelySegmentOption represents options controlling audience segments.
-type OptimizelySegmentOption int
+type OptimizelySegmentOption string
 
 const (
 	// IgnoreCache ignores cache (save/lookup)
-	IgnoreCache OptimizelySegmentOption = iota
+	IgnoreCache OptimizelySegmentOption = "IGNORE_CACHE"
 	// ResetCache resets cache
-	ResetCache
+	ResetCache OptimizelySegmentOption = "RESET_CACHE"
 )
+
+// Options defines options for controlling audience segments.
+type Options struct {
+	IgnoreCache bool
+	ResetCache  bool
+}
```

Comment:
Is this for `agent` purposes?

Reply:
yes, for Agent.
tbh I saw these in decideOptions.go, but not sure what purpose they serve.
Don't think they're needed in Agent.
Do you have thuoghts on this?
@yasirfolio3

Reply:
Actually I see Options are referred heavily in the rest of go-sdk for Decide, DecideAll calls

Reply:
`type Options struct {
	DisableDecisionEvent     bool
	EnabledFlagsOnly         bool
	IgnoreUserProfileService bool
	IncludeReasons           bool
	ExcludeVariables         bool
}`
these are used internally by SDK for `decide` api to convert user provided options to a struct for improved performance and ease of access. Basic idea is to avoid iterating through the options array again and again by converting it into a struct.

Reply:
We only provide 2 set of options for Segments, `IGNORE_CACHE` and `RESET_CACHE` so this won't make much of a difference but i'll still leave it upto you.

Reply:
Do yuo see the need for the IgnoreCache and ResetCache booleans for Options?
Looks like we are not using then on the sdk.

Reply:
What about this line: https://github.com/optimizely/go-sdk/blob/a96c3071c92185509960508cd84b83c87003aae8/pkg/client/client.go#L275 I think converting enumerated constants to string based constants is accepted here in that line above. Thoughts?

Reply:
> o.OdpManager.FetchQualifiedSegments

not really, these are not required as of now.

Reply:
> What about this line:
> 
> https://github.com/optimizely/go-sdk/blob/a96c3071c92185509960508cd84b83c87003aae8/pkg/client/client.go#L275
> 
> I think converting enumerated constants to string based constants is accepted here in that line above. Thoughts (unit tests pass)?

yep, there won't be any issue because of this: [https://github.com/optimizely/go-sdk/blob/master/pkg/odp/segment/segment_manager.go#L111](url)